packages/mobile/src/components/drawer/Drawer.tsx

```diff
@@ -506,30 +580,35 @@ export const Drawer: DrawerComponent = ({
      * When the user releases their hold of the drawer
      */
     onPanResponderRelease: (e, gestureState) => {
-      if (isOpen || isOpenToInitialOffset) {
+      if (
+        isOpen ||
+        isOpenIntent.current ||
+        (!isOpen && shouldCloseToInitialOffset)
+      ) {
         // Close if open & drag is past cutoff
         if (
           gestureState.vy > 0 &&
-          gestureState.moveY > height - MOVE_CUTOFF_CLOSE * drawerHeight
+          gestureState.moveY >
+            FULL_DRAWER_HEIGHT - MOVE_CUTOFF_CLOSE * drawerHeight
         ) {
-          if (isOpenToInitialOffset) {
-            slideOut(initialOffsetOpenPosition)
+          if (shouldCloseToInitialOffset) {
+            slideOut(initialOffsetOpenPosition, gestureState.vy, onClose)
+            isOpenIntent.current = false
             borderRadiusAnim.setValue(0)
-            onClose()
           } else {
-            slideOut(initialPosition)
-            onClose()
+            slideOut(initialPosition, gestureState.vy, onClose)
```

Comment:
so are there cases where a quick slideOut, then slideIn animation would want us to cancel the onClose call, or a guess that is handled with the "finished" boolean in the animation onFinished callback?

Reply:
precisely! the finished thing is pretty cool